absl::Container compilation issue with Visual Studio 2017 (VS v15.8)

EDIT: abseil-cpp HEAD *may* already contains a fix for this (we still have issue #193 on HEAD -> difficult to reproduce without abseil-cpp source code hack -> unreliable)

When trying to compile on windows the project google/or-tools#871 (abseil branch) which depends on abseil-cpp (commit 45221ccc4ed643e4209b0cc5798e97203f108fa8) headers, we are facing some issues with Appveyor CI.

> dependencies\\install\\include\absl/container/internal/raw_hash_map.h(73): fatal error C1001: An internal error has occurred in the compiler.
(compiler file 'msc1.cpp', line 1518)

https://github.com/abseil/abseil-cpp/blob/5b70a8910b2e6fb0ce5193a41873139a126d2f7f/absl/container/internal/raw_hash_map.h#L72-L75

Appveyor VS setup: https://www.appveyor.com/docs/windows-images-software/#visual-studio-2017
> Visual Studio Community 2017 version 15.8.5

AFAIK it seems related to [this issue](https://developercommunity.visualstudio.com/content/problem/312852/internal-compiler-error-in-vs158-msc1cpp-line-1518.html)

Appveyor Log: https://ci.appveyor.com/project/lperron/or-tools-98u1n/build/job/y08e4mqquguutnrm#L3971
```ps
cl /EHsc /MD /nologo -nologo  /D__WIN32__ /DPSAPI_VERSION=1 /DNOMINMAX /DWIN32_LEAN_AND_MEAN=1 /D_CRT_SECURE_NO_WARNINGS /O2 -DNDEBUG /I. /Iortools/gen /I"dependencies\\install\\include" /DGFLAGS_DLL_DECL= /DGFLAGS_DLL_DECLARE_FLAG= /DGFLAGS_DLL_DEFINE_FLAG= /DGOOGLE_GLOG_DLL_DECL= /DUSE_CLP  /I"dependencies\\install\\include\\coin" /DUSE_CBC /DUSE_GLOP /DUSE_BOP -c .\\ortools\\bop\\complete_optimizer.cc /Foobjs\\bop\\complete_optimizer.obj
complete_optimizer.cc
dependencies\\install\\include\absl/container/internal/raw_hash_map.h(73): fatal error C1001: An internal error has occurred in the compiler.
(compiler file 'msc1.cpp', line 1518)
```

I was able to compile using `Visual Studio 2017 v15.7.5`

Not sure if it fixable on your side or if you should add a note for VS users...

--- a/absl/container/internal/raw_hash_map.h
+++ b/absl/container/internal/raw_hash_map.h
@@ -39,11 +39,14 @@ class raw_hash_map : public raw_hash_set<Policy, Hash, Eq, Alloc> {
   using MappedConstReference = decltype(P::value(
       std::addressof(std::declval<typename raw_hash_map::const_reference>())));
 
+  using KeyArgImpl = container_internal::KeyArg<IsTransparent<Eq>::value &&
+                                                IsTransparent<Hash>::value>;
+
  public:
   using key_type = typename Policy::key_type;
   using mapped_type = typename Policy::mapped_type;
-  template <typename K>
-  using key_arg = typename raw_hash_map::raw_hash_set::template key_arg<K>;
+  template <class K>
+  using key_arg = typename KeyArgImpl::template type<K, key_type>;
 
   static_assert(!std::is_reference<key_type>::value, "");
   // TODO(alkis): remove this assertion and verify that reference mapped_type is

